src/streams.c

```diff
@@ -98,6 +98,67 @@ Int READ_COMMAND ( void )
     return 1;
 }
 
+Obj FuncREAD_ALL_COMMANDS( Obj self, Obj stream, Obj echo )
+{
+    ExecStatus status;
+    Int result_count, result_capacity;
+    Obj result, result_list;
+
+    // 16 Results should be enough for everyone
+    result_count = 1;
+    result_capacity = 16;
+    result_list = NEW_PLIST( T_PLIST, result_capacity );
+    SET_LEN_PLIST(result_list, result_count);
+
+    result = NEW_PLIST( T_PLIST, 2 );
+    SET_LEN_PLIST(result, 1);
+    SET_ELM_PLIST(result, 1, False );
+
+    SET_ELM_PLIST(result_list, 1, result);
+
+    /* try to open the file                                                */
+    if (!OpenInputStream(stream) ) {
+        return result_list;
```

Comment:
So in case of failure, you return `[[false]]` ? What is the supposed output of your function? I would appreciate a brief comment before the function which summarizes what it does, and what its output is. 

For now, my guess is that it's a a list of "output lists" similar to what e.g. `CALL_WITH_CATCH` return, i.e. either `[false]` if there was an error, or else `[true]` or `[true,result]`.

Assuming that guess is right, doesn't that mean that I cannot distinguish failing to open a stream from a stream with a single command that produced an error? Wouldn't it then be better to return `Fail` here?